### PR TITLE
Move the apiserver addresses controller into the etcd package

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

This controller only needs to run when using managed etcd, so move it in
with the rest of the etcd stuff. This change also modifies the
controller to only watch the Kubernetes service endpoint, instead of
watching all endpoints in the entire cluster.

Fixes an error message revealed by use of a newer grpc client in
Kubernetes 1.24, which logs an error when the Put to etcd failed because
kine doesn't support the etcd Put operation. The controller shouldn't
have been running without etcd in the first place.

#### Types of Changes ####

bugfix

#### Verification ####

* See linked issue
* Confirm that `k3s server` logs do not contain any mention of "managed etcd" when not using managed etcd.

#### Linked Issues ####

* #5389 

#### User-Facing Change ####
```release-note
Managed etcd controllers no longer run on clusters not using managed etcd.
```

### CODE REVIEW

**Comment by brandond** [comment]
Marking as WIP since I am not 100% on the current approach actually being correct.

**Comment by brandond** [comment]
Confirmed that everything still works as usual with this change:
```console
/ # kubectl get nodes
NAME           STATUS   ROLES                  AGE     VERSION
k3s-agent-1    Ready    <none>                 44s     v1.23.5+k3s-82ff7e2d
k3s-server-1   Ready    etcd                   7m9s    v1.23.5+k3s-82ff7e2d
k3s-server-2   Ready    etcd                   7m8s    v1.23.5+k3s-82ff7e2d
k3s-server-3   Ready    etcd                   7m12s   v1.23.5+k3s-82ff7e2d
k3s-server-4   Ready    control-plane,master   7m16s   v1.23.5+k3s-82ff7e2d
k3s-server-5   Ready    control-plane,master   6m58s   v1.23.5+k3s-82ff7e2d
```

File: pkg/etcd/apiaddresses_controller.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	"github.com/k3s-io/k3s/pkg/util"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func RegisterEndpointsHandlers(ctx context.Context, etcd *ETCD) error {
```
[the remark this answers is not in the document]
Comment (brandond) [reply pkg/etcd/apiaddresses_controller.go:16]: For some reason these are all exported. Let me go check in RKE2 to make sure we're not referencing them over there for some reason.

[the remark this answers is not in the document]
Comment (brandond) [reply pkg/etcd/apiaddresses_controller.go]: Nope, not referenced anywhere outside this package. Updated all the `Register*` functions to no longer be exported.

**Review by dereknola — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]

**Pull request merged**